addons/spreadsheet_account/models/account.py

```diff
@@ -24,6 +25,8 @@ def _get_date_period_boundaries(self, date_period, company):
             fiscal_month = int(company.fiscalyear_last_month)
             if not (fiscal_day == 31 and fiscal_month == 12):
                 year += 1
+            if company.fiscalyear_last_month == "2" and company.fiscalyear_last_day == 29:
```

Comment:
```suggestion
            if fiscal_month == 2 and fiscal_day == 29:
```
From my understanding, you always want the range as you are in the fiscal year of the "current year" so this is enough I think:
```py
            if period_type == "year":
                fiscal_day = company.fiscalyear_last_day
                fiscal_month = int(company.fiscalyear_last_month)
                max_day = calendar.monthrange(year, fiscal_month)[1]
                current = date(year, fiscal_month, min(fiscal_day, max_day))
                start, end = date_utils.get_fiscal_year(current, fiscal_day, fiscal_month)
```